keras_cv/bounding_box/converters.py

```diff
@@ -0,0 +1,125 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Converter functions for working with bounding boxes.
+
+Usually bounding boxes is a 2D Tensor with shape [batch, 4]. The second dimension
+will contain 4 numbers based on 2 different formats.  In KerasCV, we will use the
+`corners` format, which is [LEFT, TOP, RIGHT, BOTTOM].
+
+In this file, provide utility functions for manipulating bounding boxes and converting
+their formats.
+"""
+
+import tensorflow as tf
+
+
+def _center_xywh_to_corners(boxes, images=None):
+    x, y, width, height, rest = tf.split(boxes, [1, 1, 1, 1, -1], axis=-1)
+    return tf.concat(
+        [x - width / 2.0, y - height / 2.0, x + width / 2.0, y + height / 2.0, rest],
+        axis=-1,
+    )
+
+
+def _xywh_to_corners(boxes, images=None):
+    x, y, width, height, rest = tf.split(boxes, [1, 1, 1, 1, -1], axis=-1)
+    return tf.concat([x, y, x + width, y + height, rest], axis=-1)
+
+
+def _xyxy_no_op(boxes, images=None):
+    return boxes
+
+
+def _corners_to_xywh(boxes, images=None):
+    left, top, right, bottom, rest = tf.split(boxes, [1, 1, 1, 1, -1], axis=-1)
+    return tf.concat(
+        [left, top, right - left, bottom - top, rest],
+        axis=-1,
+    )
+
+
+def _corners_to_center_xywh(boxes, images=None):
+    left, top, right, bottom, rest = tf.split(boxes, [1, 1, 1, 1, -1], axis=-1)
+    return tf.concat(
+        [(left + right) / 2.0, (top + bottom) / 2.0, right - left, bottom - top, rest],
+        axis=-1,
+    )
+
+
+to_corners_converters = {
+    "xywh": _xywh_to_corners,
+    "center_xywh": _center_xywh_to_corners,
+    "xyxy": _xyxy_no_op,
+}
+
+from_corners_converters = {
+    "xywh": _corners_to_xywh,
+    "center_xywh": _corners_to_center_xywh,
+    "xyxy": _xyxy_no_op,
+}
+
+
+def transform_format(boxes, source, target, images=None):
+    """Converts bounding_boxes from one format to another.
+
+    Supported formats are:
+    - xyxy, also known as `corners` format.  In this format the first four axes
+        represent [left, top, right, bottom] in that order.
+    - xyWH.  In this format the first four axes represent [left, top, width, height].
+    - center_xyWH.  In this format the first two coordinates represent the x and y
+        coordinates of the center of the bounding box, while the last two represent
+        the width and height of the bounding box.
+
+    Formats are case insensitive.  It is recommended that you capitalize width and
+    height to maximize the visual difference between `xyWH` and `xyxy`.
```

Comment:
yeah, it is optional.   Users can just pass `xywh` - it gets lower cased anyways.  I suggested this because I thought it may improve readability, but we can remove it.

Reply:
We can remove this if we think it hurts readability